### PR TITLE
Add generator function for circulant graphs

### PR DESCRIPTION
**Pull request by lucjon**
This pull request adds a simple function for generating [circulant graphs](http://mathworld.wolfram.com/CirculantGraph.html), where every vertex is connected to those vertices with labels ±*j* of its own label, for all *j* in some set of 'jumps' *J*. A unit test is included which checks some common cases where well-known circulant graphs (such as the complete and cycle graphs) are generated.

### CODE REVIEW

File: networkx/generators/classic.py [hunk 1]
```diff
@@ -207,6 +208,22 @@ def circular_ladder_graph(n,create_using=None):
...
 
+
+def circulant_graph(n,offsets,create_using=None):
+    """Return the circulant graph Ci_n(x_1, x_2, ..., x_m) with n vertices.
+
+    Ci_n(x_1, ..., x_m) consists of n vertices 0, ..., n-1 such that the vertex
+    with label i is connected to the vertices labelled (i + x) and (i - x),
+    for all x in x_1 up to x_m, taken modulo n.
+    """
+    G=empty_graph(n,create_using)
```
Comment (jfinkels) [inline networkx/generators/classic.py]: Needs spaces around assignment operator, and after comma. Same with the line below.

**Comment by chebee7i** [comment]
Can you add a short note to the release notes?  `doc/source/reference/api_2.0.rst`

**Comment by lucjon** [comment]
Done --- thanks. :)

**Comment by chebee7i** [comment]
Great. The last thing is that need to modify the function signature (add spaces) and make the docstring look like all other docstrings we have. Minimally, you'll need a Parameters, Returns, and Examples section.

**Comment by MridulS** [comment]
@lucjon can you also resolve the merge conflicts please.

**Comment by lucjon** [comment]
I believe both of those have now been done, apologies for the delay.

**Comment by chebee7i** [comment]
I fixed the final error in 8ed04af956dd7f0c068b88c15f483dc13ab57c7b and then merged.  Thanks.

**Pull request merged**